Bluetooth: EDTT: Handle unknown response

joerchan opened the pull request:
Related to: https://github.com/EDTTool/EDTT/pull/11
I'm not sure how to handle the west manifest with the zephyr mirror.

Update HCI test application to give the same unsupported command RSP from the command complete and command status HCI events.

Enabled the tests that now are able to pass.

Remove deprecated defines and added some debug tracing that I had been using.

Fixes: #22085

joerchan committed:
Bluetooth: EDTT: Remove use of deprecated defines

Remove the use of BT_BUF_USER_DATA_MIN which has been deprecated.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: EDTT: Remove ncmd and opcode from EDTT command response

Encode all types of EDTT command response with similar structure to how
it is generated from command complete when it is generated from the
command status.
This structure will be:
2 byte EDTT command response code
2 byte EDTT length field
1 byte HCI status
N bytes of command output parameters

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: EDTT: Replace printk with logging

Replace printk with logging, add additional log statements for command
handling.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: EDTT: Enable disabled tests

Enable tests that was disabled due to test failure. Test have been
updated and are now passing.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -129,12 +138,12 @@ static int send_hci_command(u16_t opcode, u8_t param_len, u16_t response)
 		err = bt_send(buf);
 		if (err) {
-			printk("Failed to send HCI command %d (err %d)\n",
+			LOG_ERR("Failed to send HCI command %d (err %d)\n",
```
minor: log subsystem doesn't require `\n` at the end

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -129,12 +138,12 @@ static int send_hci_command(u16_t opcode, u8_t param_len, u16_t response)
 		}
 	} else {
-		printk("Failed to create buffer for HCI command %u\n", opcode);
+		LOG_ERR("Failed to create buffer for HCI command %u\n", opcode);
```
minor: log subsystem doesn't require `\n` at the end

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -197,10 +213,17 @@ static void command_status(struct net_buf *buf)
+		LOG_WRN("Not waiting for 0x(%02x) command status,"
+			" expected 0x(%02x)",
+			opcode,
+			waiting_opcode);
```
I think these fit on the same line

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -479,12 +503,12 @@ static void le_data_write(u16_t size)
 			err = bt_send(buf);
 			if (err) {
-				printk("Failed to send ACL Data (err %d)\n",
+				LOG_ERR("Failed to send ACL Data (err %d)\n",
```
minor: log subsystem doesn't require `\n` at the end

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -479,12 +503,12 @@ static void le_data_write(u16_t size)
 		} else {
 			err = -2; /* Failed to allocate data buffer */
-			printk("Failed to create buffer for ACL Data.\n");
+			LOG_ERR("Failed to create buffer for ACL Data.\n");
```
minor: log subsystem doesn't require `\n` at the end

jhedberg commented on tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c:
```diff
@@ -520,7 +544,7 @@ void main(void)
 	err = bt_enable_raw(&rx_queue);
 	if (err) {
-		printk("Bluetooth initialization failed (err %d)\n", err);
+		LOG_ERR("Bluetooth initialization failed (err %d)\n", err);
```
minor: log subsystem doesn't require `\n` at the end

jhedberg reviewed (changes requested):
Looks generally fine to me. A couple of small things I spotted.

aescolar committed:
manifest: EDTT: Update to latest version with HCI command fixes

Includes fixes to HCI commands,
which are necessary for
https://github.com/zephyrproject-rtos/zephyr/issues/22085
see:
https://github.com/EDTTool/EDTT/pull/11

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

aescolar commented:
Added DNM label due to the manifest pointing to a EDTT PR (feel free to remove whenever it is pointing back to a normal commit)
